rules/manager.go

```diff
@@ -977,55 +980,47 @@ func (m *Manager) LoadGroups(
 
 	shouldRestore := !m.restored
 
-	for _, fn := range filenames {
```

Comment:
What I'd suggest doing is just factoring out these 5 lines of code, rather than changing the overall structure of the code.

Reply:
Agree with this, I think we can get away with a smaller diff if we just factor out `rgs, errs := rulefmt.ParseFile(fn)`

We could make an interface for the function `LoadGroups(id string) rgs, errors` and that would simplify things. And then in the `LoadGroups` function itself do the validation of
```
expr, err := parser.ParseExpr(r.Expr.Value)
if err != nil {
  return nil, []error{errors.Wrap(err, fn)}
}
```

Would that work @owen-d?

Reply:
This makes sense. We're probably going to want to keep everything in the `for _, rg := range rgs.Groups` loop consistent across group loader types. If a case comes up where we need to change that we can do so, but for now the loader can just return rule groups/errors